### ISSUE TITLE
Significantly lower than expected eval accuracy on MNLI

### ISSUE THREAD

**Issue by 5donuts** [google-research/ALBERT#18]
Before ALBERT was moved to this repository, I downloaded the pre-trained ALBERT-base-2 from TFHub and used `run_classifier_sp.py` to evaluate the model on MNLI by modifying the provided `run.sh` script to execute the following instead of `run_pretraining_test`:
```bash
 python -m albert.run_classifier_sp \
    --output_dir="/path/to/output" \
    --export_dir="/path/to/export" \
    --do_eval \
    --nouse_tpu \
    --eval_batch_size=1 \
    --max_seq_length=4 \
    --max_eval_steps=3 \
    --vocab_file="/path/to/albert-base-2/assets/30k-clean.vocab" \
    --data_dir="/path/to/glue/MNLI" \
    --task_name=MNLI
```

This gave an eval accuracy of approximately 0.34, which is _significantly_ lower than the expected 0.84 discussed in the paper.

Has anyone else seen such low out-of-the-box evaluation results? Is this simply an issue with how I'm running the evaluation? If so, are there any recommendations for running evaluation to achieve better results?

**Comment by ZheyuYe** [comment]
Got disappointing results on SQuAD and CoLA as well.

**Comment by 0x0539** [comment]
Do not specify --max_seq_length=4, just use the default.
Did you do any fine-tuning on the task?

**Comment by 5donuts** [comment]
I left the `--max_seq_length=4` specified because it was in the original `run.sh` script, and did not know a better value.

I fine-tuned the model for 15 days on a Quadro P5000 and got only a 1% improvement in the eval accuracy.

**Comment by 0x0539** [comment]
The `run.sh` script is only intended for testing purposes and is not a method of reproducing results. That should probably be clearer from the filename or documentation.

`--max_seq_length=4` means that ALBERT only has 4 sequence positions. I do not expect performance to be very good under this condition. It's better not to change it. The default is 512: https://github.com/google-research/ALBERT/blob/master/run_classifier.py#L77

**Comment by 5donuts** [comment]
If `run.sh` is only intended for testing, then do you have examples showing how to reproduce the results of the paper?

The only documentation I've found is what's on TF-HUB and in `README.md`, neither of which are particularly exhaustive sources of information.

**Comment by 0x0539** [comment]
Ah, good point! We needed to update the docs with specific, working examples.
Please see the latest changes to `README.md`: https://github.com/google-research/ALBERT/commit/5bf155334a169f53759e3e591ddb0744b2dc9801
Hopefully that resolves the issue. Let me know if there's anything else.

**Comment by MichaelZhouwang** [comment]
I've followed the hyper parameters presented in the paper but only got accuracy of ~83.0. Have you figured out how to get the reported result? Thanks!
